code/game/machinery/suit_storage_unit.dm

```diff
@@ -187,263 +409,134 @@
 	return
 
 
-/obj/machinery/suit_storage_unit/Topic(href, href_list) //I fucking HATE this proc
+/obj/machinery/suit_storage_unit/Topic(href, href_list)
 	if(..())
 		return 1
 	if((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
 		usr.set_machine(src)
 		if(href_list["toggleUV"])
-			src.toggleUV(usr)
-			src.updateUsrDialog()
-			src.update_icon()
+			toggleUV(usr)
+			updateUsrDialog()

```

Comment:
These `updateUsrDialog()` and `update_icon()` can both go at the end of this proc to save on space

Reply:
will go for invalid options then too, being being dirty with topic calls.